s3fs/core.py

```diff
@@ -185,7 +190,7 @@ def ls(self, path, detail=False):
         try:
             files = self._ls(path)
         except ClientError:
-            files = []
+            raise FileNotFoundError(path)
```

Comment:
Things like this could use tests, ideally in the same commit where the functionality was added.